test/plots/rectanglePlotTests.ts

```diff
@@ -2,511 +2,486 @@
-        let dataset = new Plottable.Dataset(DATA);
-        gridPlot.addDataset(dataset)
-          .attr("fill", (d) => d.magnitude, colorScale);
-         gridPlot.x((d: any) => d.x, xScale)
-          .y((d: any) => d.y, yScale);
-        gridPlot.renderTo(svg);
-
         let dummyDataset = new Plottable.Dataset([]);
-        let allCells = gridPlot.selections([dataset, dummyDataset]);
+        let allCells = plot.selections([dataset, dummyDataset]);
         assert.strictEqual(allCells.size(), 4, "all cells retrieved");
         let selectionData = allCells.data();
-        assert.includeMembers(selectionData, DATA, "data in selection data");
-
+        assert.includeMembers(selectionData, data, "data in selection data");
         svg.remove();
       });
 
     });
 
-    describe("Rectangle Plot With Labels", () => {
-      let svg: d3.Selection<void>;
-      let rectanglePlot: Plottable.Plots.Rectangle<number, number>;
-      let DATA: [any];
+    describe("Label behavior", () => {
+      let plot: Plottable.Plots.Rectangle<number, number>;
+      let data = [
+        { x: 0, y: 0, x2: 1, y2: 1, val: "1" },
+        { x: 0, y: 1, x2: 1, y2: 2, val: "2" }
+      ];
       let dataset: Plottable.Dataset;
       let xScale: Plottable.Scales.Linear;
       let yScale: Plottable.Scales.Linear;
+
       beforeEach(() => {
         xScale = new Plottable.Scales.Linear();
         yScale = new Plottable.Scales.Linear();
-        svg = TestMethods.generateSVG(SVG_WIDTH / 2, SVG_HEIGHT);
-        rectanglePlot = new Plottable.Plots.Rectangle<number, number>();
-        DATA = [
-          { x: 0, y: 0, x2: 1, y2: 1, val: "1" },
-          { x: 0, y: 1, x2: 1, y2: 2, val: "2" }
-        ];
-        dataset = new Plottable.Dataset(DATA);
-        rectanglePlot.addDataset(dataset);
-        rectanglePlot.x((d: any) => d.x, xScale)
-                     .y((d: any) => d.y, yScale)
-                     .x2((d: any) => d.x2)
-                     .y2((d: any) => d.y2)
-                     .label((d: any) => d.val)
-                     .renderTo(svg);
+        plot = new Plottable.Plots.Rectangle<number, number>();
+        dataset = new Plottable.Dataset(data);
+        plot.addDataset(dataset);
+        plot.x((d: any) => d.x, xScale)
+          .y((d: any) => d.y, yScale)
+          .x2((d: any) => d.x2)
+          .y2((d: any) => d.y2)
+          .label((d: any) => d.val);
       });
 
-      it("rectangle labels disabled by default", () => {
+      it("displays rectangle labels by default", () => {
+        let svg = TestMethods.generateSVG(150, 300);
+        plot.renderTo(svg);
         let texts = svg.selectAll("text")[0].map((n: any) => d3.select(n).text());
         assert.lengthOf(texts, 0, "by default, no labels are drawn");
         svg.remove();
       });
 
-      it("rectangle labels render properly", () => {
-        rectanglePlot.labelsEnabled(true);
+      it("renders correct number of labels with the correct text", () => {
```

Comment:
Not really. This was supposed to be renders correctly, but I wanted to be more explicit. Whenever we check for the correct text, we should check for the correct number of labels, otherwise we run into the problem of `assert for all (nothing) === true`. My suggestion, I will delete the "correct number of labels bit from the test"